core/src/main/java/org/pentaho/platform/genericfile/providers/repository/RepositoryFileProvider.java

```diff
@@ -114,10 +114,24 @@ protected RepositoryFileTree getFolderTreeCore( @NonNull GetTreeOptions options
 
     FileService fileService = new FileService();
 
+    String filter;
+
+    switch ( options.getTreeFilter() ) {
+      case FOLDERS:
+        filter = "*|FOLDERS";
+        break;
+      case FILES:
+        filter = "*|FILES";
+        break;
+      case ALL:
+      default:
+        filter = "*";
+    }
+
```

Comment:
Consider creating a helper protected method to convert the filter to a repository filter string.

Reply:
I opted to go for adding a `repositoryFilterString` to the enum to centralize the filters... was this a good idea? :D

Reply:
Interesting. As discussed, the repository concept does leak into some doclets, but otherwise is not part of the API. The repository has a priviledged path, starting with `/`. That said, the repository has its own `IGenericFileProvider` implementation which supposedly concentrates all that is specific to the repository.

If we would be generalizing the generic file get tree filter to include a file name pattern (e.g. to support filtering by extension), then this would be the way, aligning to how the repository API did it, trying to minimize differences lacking a reason to do it differently. We wouldn't call it `repositoryFilterString` in that case.

However, until such a feature is developer, I'd prefer keeping this logic private to the repository implementation.